src/config.ts

```diff
@@ -133,7 +133,7 @@ export interface RangeConfig {
    */
   symbol?: string[];
 
-  [name: string]: RangeConfigValue;
+  [name: string]: RangeConfigValue | undefined;
```

Comment:
@domoritz  -- I wonder if the right thing to do is adding `?` to `name`?

Reply:
Yes, add ? instead.

Reply:
That's not valid TypeScript syntax. What about this?

```ts
export interface RangeConfig extends Partial<Record<string, RangeConfigValue>> {
    /**
     * Default range for _nominal_ (categorical) fields.
     */
    category?: string[] | VgScheme;
    /**
     * Default range for diverging _quantitative_ fields.
     */
    diverging?: string[] | VgScheme;
    /**
     * Default range for _quantitative_ heatmaps.
     */
    heatmap?: string[] | VgScheme;
    /**
     * Default range for _ordinal_ fields.
     */
    ordinal?: string[] | VgScheme;
    /**
     * Default range for _quantitative_ and _temporal_ fields.
     */
    ramp?: string[] | VgScheme;
    /**
     * Default range palette for the `shape` channel.
     */
    symbol?: string[];
}
```

Reply:
`[name: string]?: RangeConfigValue` is not?

Reply:
No.

Reply:
Ahh, I misremembered and confused it with this:

```ts
type Bar = 'a' | 'b';

export type Foo = {
  [C in Bar]?: boolean
};
```

How come that one works but not the other?

Anyway, looking at this again, it doesn't really matter. `[name: string]: RangeConfigValue` already means that the object can have no props defined.

Reply:
Here are some experiments: https://www.typescriptlang.org/play/#src=type%20Foo1%20%3D%20%7B%0A%20%20%5Bx%3A%20string%5D%3F%3A%20boolean%0A%7D%0A%0Atype%20Bar%20%3D%20'a'%20%7C%20'b'%3B%0A%0Atype%20Foo2%20%3D%20%7B%0A%20%20%5BC%20in%20Bar%5D%3F%3A%20boolean%0A%7D%3B%0A